src/Campaigns/Actions/PreventDeleteDefaultForm.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\Models\Campaign;
+
+/**
+ * @unreleased
+ */
+class PreventDeleteDefaultForm
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke($postId)
+    {
+        if (get_post_type($postId) !== 'give_forms') {
+            return;
+        }
+
+        $campaign = Campaign::findByFormId($postId);
+
+        if ($campaign && $campaign->defaultFormId == $postId) {
+            wp_die(sprintf(__('The form %s with ID %d cannot be deleted because it is the default form for a campaign.',
```

Comment:
Does this return a response to the user in the UI or does it just cause a 500 error if someone tries to trash a default form?

Reply:
@jonwaldstein I know this is not the best UX, but as a note, when we try to delete the default form using the bulk actions of the Forms tab on the campaign details page, there is also no return in the UI. The only difference here is that we are returning a 500 error.

Reply:
Okay thanks, I think we need to come up with a way to have the UI respond for customers or else this could be confusing for them 😏

Reply:
@jonwaldstein Do you think a simple alert would be enough? Since this page is "hidden" and we are recommending not to use it anymore, I think it would be enough for now.

Reply:
ya that would work 👍

Reply:
Resolved in 9e9e96a678d438935ec0fa12f66490b665cdd29f